src/txn/txn_recover.c

```diff
@@ -321,6 +320,15 @@ __txn_log_recover(WT_SESSION_IMPL *session,
 	/* First, peek at the log record type. */
 	WT_RET(__wt_logrec_read(session, &p, end, &rectype));
 
+	/*
+	 * Record the highest LSN we process during the metadata phase.
+	 * If not the metadata phase, then stop at that LSN.
+	 */
+	if (r->metadata_only)
+		r->max_rec_lsn = *next_lsnp;
+	else if (__wt_log_cmp(lsnp, &r->max_rec_lsn) > 0)
```

Comment:
Should this comparison be `if (__wt_log_cmp(lsnp, &r->max_rec_lsn) >= 0)` instead of `... > 0)` ?